schedule_lessons/dashboard/static/dashboard/js/settings.js

```diff
@@ -0,0 +1,88 @@
+$(document).ready(() => {
+
+  $('#secondNavBtn').click(() => {
+    $('#navbarSupportedContent').slideToggle('fast');
+  });
+
+  $('#deleteAccountConfirm').click(() => {
+    $('#deleteAccountModal').modal('hide');
+    $.ajax({
+      type: 'DELETE',
+      url: '/dashboard/my_profile/delete_account/',
+      success: function () {
+        document.location.href = "/";
+      }
+    });
+  });
+
+  $("#confirmChangePassword").click(() => {
+    $.ajax({
+      type: 'POST',
+      url: '/dashboard/my_profile/change_password/',
+      data: { 'old_password': $("#oldPasswordInput").val(), 'new_password1': $("#newPasswordInput1").val(), 'new_password2': $("#newPasswordInput2").val() },
```

Comment:
For the first point, from what I've read, keys need to be defined as strings. For the second point, that's a good point. For simple errors like empty forms, frontend checking makes more sense. I actually think we should have both frontend and backend checking. Frontend for the simple stuff and backend for more complicated backend checking. I started an issue on this #141. If you have good resources to share on error handling in webapps, pop them in there.

Reply:
all the checks should happen in both backend and frontend
frontend for user experience improvements and also reducing wait times for server responses
backend for security and actual functionality purposes in case a user tries to make a mock request using curl or something like that to an endpoint without actually using the frontend and tries to change someone's password or delete someone's account